scripts/compress_images_test.py

```diff
@@ -0,0 +1,238 @@
+        self.assertEqual(compression_type, 'Zip')
+
+    def test_get_compression_type_lzw_extension(self) -> None:
+        """Test get_compression_type returns 'LZW' for JPG."""
+        compressor = compress_images.ImageCompressor(self.temp_dir)
+        compression_type = compressor.get_compression_type('.jpg')
+        self.assertEqual(compression_type, 'LZW')
+
+    def test_compress_images(self) -> None:
+        """Test image compression process."""
+        with self.swap(subprocess, 'run', mock.MagicMock()):
+            self.assertFalse(os.path.exists(self.output_dir))
+            result_images: List[CompressedImageInfo] = [
+                {
+                    'path': pathlib.Path(self.png_path),
+                    'original_size': 1000,
+                    'new_size': 500
+                }
+            ]
+            compressor = compress_images.ImageCompressor(
+                self.temp_dir,
+                output_dir=self.output_dir
+            )
+            compressor.compress_images(result_images)
+
+            self.assertTrue(os.path.exists(self.output_dir))
+
+            rel_path = os.path.relpath(self.png_path)
+            output_file_path = pathlib.Path(self.output_dir) / rel_path
+            self.assertTrue(output_file_path.exists())
+
+            if pathlib.Path(self.png_path).exists():
+                self.assertEqual(
+                    output_file_path.stat().st_size,
+                    pathlib.Path(self.png_path).stat().st_size
+                )
+
+    def test_compress_images_directory_removal_error(self) -> None:
+        """Test error handling when removing output directory fails."""
+        compressor = compress_images.ImageCompressor(
+            self.temp_dir,
+            output_dir=self.output_dir
+        )
+        os.makedirs(self.output_dir, exist_ok=True)
+        mock_rmtree = mock.MagicMock(ignore_errors=OSError('Permission denied'))
+        mock_log_error = mock.MagicMock()
+        mock_subprocess_run = mock.MagicMock(
+            return_value=mock.MagicMock(
+                returncode=1,
+                stderr='Permission denied error'
+            )
+        )
+
+        with self.swap(shutil, 'rmtree', mock_rmtree):
+            with self.swap(logging, 'error', mock_log_error):
+                with self.swap(subprocess, 'run', mock_subprocess_run):
+                    result_images: List[CompressedImageInfo] = [
+                        {
+                            'path': pathlib.Path(self.png_path),
+                            'original_size': 1000,
+                            'new_size': 500
+                        }
+                    ]
+
+                    compressor.compress_images(result_images)
+
+        mock_rmtree.assert_called_once_with(self.output_dir, ignore_errors=True)
+        mock_log_error.assert_called_with(
+            '[ERROR]: %s occurred on file %s',
+            pathlib.Path(self.png_path),
+            'Permission denied error'
```

Comment:
this is a constant, so you can put it into the first argument. Doing so will also reveal that the arguments are being passed to `logging.error` in the wrong order (I left a separate comment about that)